tests/Functional/Schema/PostgreSQLSchemaManagerTest.php

```diff
@@ -598,6 +599,56 @@ public static function autoIncrementTypeMigrations(): iterable
             'bigint->int' => ['bigint', 'integer', 'INT'],
         ];
     }
+
+    public function testPartitionTable(): void
+    {
+        $this->connection->executeStatement('DROP TABLE IF EXISTS partitioned_table');
+        $this->connection->executeStatement(
+            'CREATE TABLE partitioned_table (id INT PRIMARY KEY) PARTITION BY LIST (id);',
+        );
+        $this->connection->executeStatement('CREATE TABLE partition PARTITION OF partitioned_table DEFAULT;');
+        try {
+            $this->schemaManager->introspectTable('partition');
+        } catch (TableDoesNotExist $e) {
+        }
+
+        self::assertNotNull($e ?? null, 'Partition table should not be introspected');
+
+        $tableFrom = $this->schemaManager->introspectTable('partitioned_table');
+
+        $tableTo = $this->schemaManager->introspectTable('partitioned_table');
+        $tableTo->addColumn('foo', Types::INTEGER);
+
+        $platform = $this->connection->getDatabasePlatform();
+        $diff     = $this->schemaManager->createComparator()->compareTables($tableFrom, $tableTo);
+
+        $sql = $platform->getAlterTableSQL($diff);
+        self::assertSame(['ALTER TABLE partitioned_table ADD foo INT NOT NULL'], $sql);
+
+        $this->schemaManager->alterTable($diff);
+
+        $tableFinal = $this->schemaManager->introspectTable('partitioned_table');
+        self::assertTrue($tableFinal->hasColumn('id'));
+        self::assertTrue($tableFinal->hasColumn('foo'));
+
+        $partitionedTableStillExists = (bool) ($this->connection->executeQuery(
+            "select count(*) as count from pg_class where relname = 'partitioned_table' and relkind = 'p'",
+        )->fetchAssociative()['count'] ?? null);
+        self::assertTrue($partitionedTableStillExists);
+
+        $partitionStillExists = (bool) ($this->connection->executeQuery(
+            <<<'SQL'
+            select count(*) as count
+            from pg_class parent
+            inner join pg_inherits on pg_inherits.inhparent = parent.oid
+            inner join pg_class child on pg_inherits.inhrelid = child.oid 
+                and child.relkind = 'r'
+                and child.relname = 'partition'
+            where parent.relname = 'partitioned_table' and parent.relkind = 'p';
+            SQL,
+        )->fetchAssociative()['count'] ?? null);
+        self::assertTrue($partitionStillExists);
```

Comment:
We know exactly how many partitions to expect, don't we?

```suggestion
        $partitionsCount = (int) ($this->connection->fetchOne(
            <<<'SQL'
            select count(*) as count
            from pg_class parent
            inner join pg_inherits on pg_inherits.inhparent = parent.oid
            inner join pg_class child on pg_inherits.inhrelid = child.oid 
                and child.relkind = 'r'
                and child.relname = 'partition'
            where parent.relname = 'partitioned_table' and parent.relkind = 'p';
            SQL,
        );
        self::assertSame(1, $partitionsCount);
```

Reply:
done the same for both requests